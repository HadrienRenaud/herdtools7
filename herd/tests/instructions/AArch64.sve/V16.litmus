AArch64 V16
{
 int x[4]={0,1,2,3};
 int y[4]={5,6,7,8};
 0:X0=x;
 0:X1=y;
}
P0                                ;
MOV W2,#0                         ;
MOV W3,#4                         ;
WHILELT P0.S,W2,W3                ;
LD1W {Z1.S},P0/Z,[X0]             ;
LD1W {Z2.S},P0/Z,[X1,Z1.S,UXTW #2];
forall 0:V1.4S={0,1,2,3} /\ 0:V2.4S={5,6,7,8}

