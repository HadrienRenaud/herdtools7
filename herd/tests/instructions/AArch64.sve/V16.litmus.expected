Test V16 Required
States 1
0:V1.4S={0,1,2,3}; 0:V2.4S={5,6,7,8};
Ok
Witnesses
Positive: 1 Negative: 0
Flag Scalable-Vector-Extensions-is-work-in-progress
Condition forall (0:V1.4S={0,1,2,3} /\ 0:V2.4S={5,6,7,8})
Observation V16 Always 1 0
Hash=97b55b740335168c5e12a073693d253a

